src/pages.c

```diff
@@ -491,5 +501,19 @@ pages_boot(void) {
 
 	init_thp_state();
 
+	/* Detect lazy purge runtime support. */
+	if (pages_can_purge_lazy) {
+		bool committed = false;
+		void *madv_free_page = os_pages_map(NULL, PAGE, PAGE, &committed);
+		if (madv_free_page == NULL) {
+			return true;
+		}
+		assert(pages_can_purge_lazy_runtime);
```

Comment:
It seems like this just asserts that pages_boot() is only called once?

Reply:
Yeah. Added it mostly to make the code easier to follow.